include/ddc/kernels/splines/splines_linear_problem_2x2_blocks.hpp

```diff
@@ -0,0 +1,318 @@
+    {
+        assert(i < size());
+        assert(j < size());
+
+        std::size_t const nq = m_top_left_block->size();
+        if (i < nq && j < nq) {
+            return m_top_left_block->get_element(i, j);
+        } else if (i >= nq && j >= nq) {
+            return m_bottom_right_block->get_element(i - nq, j - nq);
+        } else if (j >= nq) {
+            return m_top_right_block(i, j - nq);
+        } else {
+            return m_bottom_left_block(i - nq, j);
+        }
+    }
+
+    virtual void set_element(std::size_t const i, std::size_t const j, double const aij) override
+    {
+        assert(i < size());
+        assert(j < size());
+
+        std::size_t const nq = m_top_left_block->size();
+        if (i < nq && j < nq) {
+            m_top_left_block->set_element(i, j, aij);
+        } else if (i >= nq && j >= nq) {
+            m_bottom_right_block->set_element(i - nq, j - nq, aij);
+        } else if (j >= nq) {
+            m_top_right_block(i, j - nq) = aij;
+        } else {
+            m_bottom_left_block(i - nq, j) = aij;
+        }
+    }
+
+private:
+    // @brief Compute the Schur complement delta - lambda*Q^-1*gamma.
+    virtual void compute_schur_complement()
+    {
+        Kokkos::parallel_for(
+                "compute_schur_complement",
+                Kokkos::MDRangePolicy<Kokkos::DefaultHostExecutionSpace, Kokkos::Rank<2>>(
+                        {0, 0},
+                        {m_bottom_right_block->size(), m_bottom_right_block->size()}),
+                [&](const int i, const int j) {
+                    double val = 0.0;
+                    for (int l = 0; l < m_top_left_block->size(); ++l) {
+                        val += m_bottom_left_block(i, l) * m_top_right_block(l, j);
+                    }
+                    m_bottom_right_block
+                            ->set_element(i, j, m_bottom_right_block->get_element(i, j) - val);
+                });
+    }
+
+public:
+    /**
+     * @brief Perform a pre-process operation on the solver. Must be called after filling the matrix.
+     *
+     * Block-LU factorize the matrix A according to the Schur complement method. The block-LU factorization is:
+     *
+     * A = |   Q    |             0             |  | I | Q^-1*gamma |
+     *     | lambda | delta - lambda*Q^-1*gamma |  | 0 |      I     |
+     *
+     * So we perform the factorization inplace to store only the relevant blocks in the matrix (while factorizing
+     * the blocks themselves if necessary):
+     *
+     * |   Q    |         Q^-1*gamma        |
+     * | lambda | delta - lambda*Q^-1*gamma |
+     */
+    void setup_solver() override
+    {
+        m_top_left_block->setup_solver();
+
+        // Compute Q^-1*gamma
+        auto top_right_block_device = create_mirror_view_and_copy(ExecSpace(), m_top_right_block);
+        m_top_left_block->solve(top_right_block_device);
+        deep_copy(m_top_right_block, top_right_block_device);
```

Comment:
Could you move these lines inside `compute_schur_complement` ?

Reply:
Yes but it would not only compute the schur complement anymore ? It's a detail though

Reply:
Well the documentation of the `compute_schur_complement` says
> // @brief Compute the Schur complement delta - lambda*Q^-1*gamma.

But the computation `Q^-1*gamma` seems to be handled outside of the function ?

Reply:
Yes because `Q^-1*gamma` is one of the blocks of the matrix whereas Schur complement is another block

Reply:
Ok I think i get it. So `compute_schur_complement` assumes that the top right block already contains `Q^-1 gamma`

Reply:
Yes exactly